scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +657,119 @@ def __mul__(self, other):
         if self._single and other._single:
             result = result[0]
         return self.__class__(result, normalized=True)
+
+    def __getitem__(self, val):
+        """Extract rotation at given index(es) from object.
+
+        This function returns a new `Rotation` instance containing:
+
+            - a single rotation, if `val` is a single index
+            - a stack of rotation(s), if `val` is a slice, or an index array.
+              In cases where a single index is ultimately specified, the stack
+              will contain a single rotation.
+
+        Parameters
+        ----------
+        val : index, slice, or index array
+            This parameter specifies which rotation(s) to extract.
+        """
+        # __init__ now copies for all cases
+        return self.__class__(self._quat[val], normalized=True)
+
+    def copy(self):
+        """Copy this object.
+
+        Returns a copy of this `Rotation` instance.
+        """
+        # __init__ now copies for all cases
+        return self.__class__(self.as_quaternion(), normalized=True)
+
+
+class Slerp(object):
+    """Spherical Linear Interpolation of Rotations.
+
+    The interpolation between consecutive rotations is performed as a rotation
+    around a fixed axis with a constant angular velocity [1]_. This ensures
+    that the interpolated rotations follow the shortest path between initial
+    and final orientations.
+
+    Parameters
+    ----------
+    times : array_like, shape (N,)
+        Times of the known rotations. At least 2 times must be specified.
+    rotations : `Rotation` instance
+        `N` rotations to perform the interpolation between. At least 2
+        rotations must be specified.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
+    """
+    def __init__(self, times, rotations):
+        # We don't need to store all rotations, just rotations[:-1] and
+        # interpolation rotvecs
+        if rotations.num_rots == 1:
+            raise ValueError("Expected at least 2 rotations in object, "
+                             "got {}.".format(rotations.num_rots))
```

Comment:
I suppose got 1 ) I guess you don't need to say that.

So "`rotations` must contain at least 2 rotations.`